packages/pancake-uikit/src/hooks/useTooltip/useTooltip.tsx

```diff
@@ -21,18 +21,48 @@ const useTooltip = (
   const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
 
   const [visible, setVisible] = useState(false);
+  const tooltipHoverRef = useRef(false);
+  const tooltipHoverTimeoutRef = useRef<number>();
 
-  const hideTooltip = useCallback((e: Event) => {
-    e.stopPropagation();
-    e.preventDefault();
-    setVisible(false);
-  }, []);
-
-  const showTooltip = useCallback((e: Event) => {
-    e.stopPropagation();
-    e.preventDefault();
-    setVisible(true);
-  }, []);
+  const hideTooltip = useCallback(
+    (e: Event) => {
+      const hide = () => {
+        e.stopPropagation();
+        e.preventDefault();
+        setVisible(false);
+      };
+
+      if (trigger === "hover") {
+        if (e.target === targetElement) {
+          clearTimeout(tooltipHoverTimeoutRef.current);
+          tooltipHoverTimeoutRef.current = window.setTimeout(() => {
+            if (!tooltipHoverRef.current) {
```

Comment:
This might create situation that when you leave the target element and then re-enter the tooltip will hide, even though you're still hovering over the target element.

Reply:
Fixed